invertedindex/ds_index.go

```diff
@@ -29,20 +30,20 @@ func NewDataStoreIndex(d ds.Batching) *DataStoreIndex {
 	}
 }
 
-func (d *DataStoreIndex) AddCidsForShard(cidIter CidIterator, s shard.Key) error {
+func (d *DataStoreIndex) AddMultihashesForShard(mhIter index.IterableIndex, s shard.Key) error {
 	batch, err := d.ds.Batch()
 	if err != nil {
 		return fmt.Errorf("failed to create ds batch: %w", err)
 	}
 
-	for _, c := range cidIter {
-		ck := ds.NewKey(c.String())
+	err = mhIter.ForEach(func(mh multihash.Multihash, _ uint64) error {
+		ck := ds.NewKey(mh.HexString())
```

Comment:
would be best to have a multiHashToDsKey function that all CRUD operations can use in this file.